packages/babel-parser/src/parser/expression.js

```diff
@@ -2637,4 +2660,39 @@ export default class ExpressionParser extends LValParser {
 
     return ret;
   }
+
+  // https://github.com/tc39/proposal-js-module-blocks
+  parseModuleExpression(): N.ModuleExpression {
+    this.expectPlugin("moduleBlocks");
+    const node = this.startNode<N.ModuleExpression>();
+    this.next(); // eat "module"
+    const oldLabels = this.state.labels;
+    this.state.labels = [];
+    const oldExportedIdentifiers = this.state.exportedIdentifiers;
+    this.state.exportedIdentifiers = [];
+    this.eat(tt.braceL);
```

Comment:
If we use `lookaheadCharCode` this should then become `this.expect`, otherwise it could be a `{|` token.

Reply:
🤦‍♂️Oh I hope `{|` never be chosen for record and tuple proposal.